### PR TITLE
[WIP] Support exporting to ONNX

### PR DESCRIPTION
**Pull request by yg2288**
This pull request is in reference to issue #18

### CODE REVIEW

File: dlpy/model.py [hunk 1]
```diff
@@ -2342,30 +2342,114 @@ def save_to_table_with_caslibify(self, path):
```
Comment (ethem-kinginthenorth) [inline dlpy/model.py]: did you run the unittests? are they clean? could you please also attach the results of it?

File: dlpy/model.py [hunk 2]
```diff
@@ -2342,30 +2342,114 @@ def save_to_table_with_caslibify(self, path):
...
+            model weights table.  Only effective when
+            output_format='onnx'.  If no csv file is specified when 
+            deploying to ONNX, the weights will be fetched from the 
+            CAS server.  This may take a long time to complete if 
+            the size of model weights is large.
 
         Notes
         -----
-        Currently, this function only supports sashdat and astore formats.
+        Currently, this function supports sashdat, astore, and onnx formats.
```
Comment (ethem-kinginthenorth) [inline dlpy/model.py]: can you also add links for onnx here?

File: dlpy/model_conversion/write_onnx_model.py [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+    if isinstance(model_weights, pd.DataFrame):
+        fetch = False
+    else:
+        fetch = True
+
+    model_name = model_table.query('_DLKey1_ = "modeltype"') \
+                            .fetch()['Fetch']['_DLKey0_'][0]
+
+    for layer in layers:
+        if layer.__class__.__name__.lower() == 'inputlayer':
```
Comment (ethem-kinginthenorth) [inline dlpy/model_conversion/write_onnx_model.py]: there should be something like this layer.type() == 'inputlayer', can you check it out?

File: dlpy/model.py (hunk 1 above)
Comment (yg2288) [reply dlpy/model.py]: test results: https://gist.github.com/yg2288/a6e1bf2a0c6e62256a67c5cb765fef65  

- missing the 'giraffe_dolphin_small' data folder
- seems to be a path error in test_audio_1
- rest are good (I don't have timedata action set, so couldn't run those)

File: dlpy/model.py (hunk 2 above)
Comment (yg2288) [reply dlpy/model.py]: added link to onnx

File: dlpy/model.py [hunk 4]
```diff
@@ -2342,30 +2342,114 @@ def save_to_table_with_caslibify(self, path):
...
+            Specifies the client-side path of the csv file of the 
+            model weights table.  If no csv file is specified, the 
+            weights will be fetched from the CAS server.  This can 
+            take a long time to complete if the size of model weights
+            is large.
+
+        '''
+
+        from .model_conversion.write_onnx_model import sas_to_onnx
+        if model_weights is None:
```
[the remark this answers is not in the document]
Comment (yg2288) [reply dlpy/model.py]: added check for castable

File: dlpy/model_conversion/write_onnx_model.py (hunk 3 above)
Comment (yg2288) [reply dlpy/model_conversion/write_onnx_model.py]: changed to use layer.type

File: dlpy/model.py [hunk 5]
```diff
@@ -2668,11 +2752,19 @@ def extract_conv_layer(layer_table):
...
         del conv_layer_config['trunc_fact']
 
-    if layer_table['_DLNumVal_'][layer_table['_DLKey1_'] == 'convopts.no_bias'].any():
+    dl_numval = layer_table['_DLNumVal_']
+    if dl_numval[layer_table['_DLKey1_'] == 'convopts.no_bias'].any():
         conv_layer_config['include_bias'] = False
     else:
         conv_layer_config['include_bias'] = True
 
+    padding_width = dl_numval[layer_table['_DLKey1_'] == 'convopts.pad_left'].tolist()[0]
```
[the remark this answers is not in the document]
Comment (yg2288) [reply dlpy/model.py]: checked this for compatibility

**Pull request merged**